addon/components/course-materials.js

```diff
@@ -1,126 +1,156 @@
 import Component from '@ember/component';
-import { Promise as RSVPPromise, map } from 'rsvp';
 import EmberObject, { computed } from '@ember/object';
-import { isPresent, isEmpty } from '@ember/utils';
-import SortableTable from 'ilios-common/mixins/sortable-table';
+import { reads } from '@ember/object/computed';
+import { isEmpty, isPresent } from '@ember/utils';
+import { all, task, timeout } from 'ember-concurrency';
 import layout from '../templates/components/course-materials';
+import DS from 'ember-data';
 
-export default Component.extend(SortableTable, {
+const { PromiseArray } = DS;
+
+const DEBOUNCE_DELAY = 250;
+
+export default Component.extend({
   layout,
-  course: null,
-  sortBy: null,
+
   classNames: ['course-materials'],
-  typesWithUrl: null,
-  filter: '',
-  sessionLearningMaterials: computed('course.sessions.[]', function(){
-    const course = this.get('course');
-    return new RSVPPromise(resolve => {
-      course.get('sessions').then(sessions => {
-        map(sessions.toArray(), session => {
-          return session.get('learningMaterials');
-        }).then(learningMaterials => {
-          let flat = learningMaterials.reduce((flattened, obj) => {
-            return flattened.pushObjects(obj.toArray());
-          }, []);
-
-          resolve(flat);
-        });
-      });
-    });
+
+  clmQuery: '',
```

Comment:
Since this is the external API of this component it would be clearer to define `clmQuery` and `slmQuery` as `courseQuery` in the same way as `courseSort`.

Reply:
👍